bindings/nodejs/src/execution_result.rs

```diff
@@ -27,7 +26,11 @@ impl JsExecutionResult {
     pub(crate) async fn to_arrow_inner(&mut self) -> napi::Result<Vec<u8>> {
         let res = match &mut self.0 {
             ExecutionResult::Query { stream, .. } => {
-                let batch = stream.next().await.unwrap().map_err(JsGlareDbError::from)?;
+                let batch = stream.next().await;
+                if batch.is_none() {
+                    return Ok(vec![]);
+                }
+                let batch = batch.unwrap().map_err(JsGlareDbError::from)?;
```

Comment:
There's no need to check that we have at least one batch, we should just be iterating over the stream and writing all batches that we get to the writer.

The schema can be gotten by calling `stream.schema()`, we don't need to the first batch.

Reply:
For this operation, we always expect it to be a single batch. Since we are writing it to arrow file format, it always needs to be within a single batch anyways.

Reply:
Now that I think about it.. Until we have native handling of recordbatches, we should probably just coalesce all plans to a single record batch.

Reply:
Just realized what you were suggesting. Ignore previous comment.

Reply:
> For this operation, we always expect it to be a single batch.

This isn't the case. The results of a query very much could result in a stream of multiple batches. All our output operations (psql, python) should be handling the case where there's multiple record batches on the stream.

> Since we are writing it to arrow file format, it always needs to be within a single batch anyways.

The IPC format can handle multiple batches (by just calling `write` more than once). And the arrow js side should be able to load a table backed by multiple batches (if it doesn't that's a pretty significant bug on their end since IPC is very much expected to be storing multiple batches).

Reply:
yeah sorry, I originally misunderstood your comment. Just pushed up some changes that should properly handle the conversion.